### DISCUSSION TITLE
How to iterate over test values with Z being one of them?

### DISCUSSION THREAD

**Discussion by clifcox** [hneemann/Digital#1099] [Q&A]
Greetings folks, :-)

I'm intrigued that Digital has such great support for testing high Z inputs and outputs. For example relabeling an input 'in' as 'in_out' apparently lets you test what value it's getting from other connections. Cool!

Though I'm wondering if there is a way to programmatically decide if a pin gets a 0, 1, or Z value? For example Z could be a digit, or perhaps it could be assigned to a variable, or even returned as part of an ite( cond, ...) expression?

Thanks for your help :-)
Clif

**Comment by hneemann** [comment]
Do you mean something like this?

[z.dig.zip](https://github.com/hneemann/Digital/files/10418733/z.dig.zip)

**Comment by clifcox** [comment]
Not exactly... I would like to have an expression for a pin in a test row that could evaluate to Z in the same way that
I could assign a value of 0 or 1 to a bit, I would also like the option of assigning a 'Z' to it. For example:

let val = 0;
let val = 'Z';

0 1 (val)

OR

0 1 (ite( n < 4, 'Z', (n)))
